EmerisHQ/demeris
# feat: dex agg 4

## Description

Please check testing section

Fixes: #1518 
Fixes: #1594
Fixes: #1595 
Fixes: #1596 

## Feature flags

VITE_FEATURE_DEX_AGG=true
VITE_FEATURE_ENABLE_SWAP_ROUTE=true

## Testing

1518: 
1. Select pay, receive denoms and add input amount
2. Click on the dex and then click on 'x transactions'
3. Check subtitle (x transactions over y chains) and **also ensure that the route is properly constructed cross checking with the API call in the network tab.**

1595: 
1. Ensure the API call is debounced and the attached gif in the issue is not an issue. 😛 

1596:
1. Use feature flag `VITE_FEATURE_ENABLE_SWAP_ROUTE` to enable/disable swap route.
2. Clicking on x transactions in a quote should have no effect when disabled.

## Additional Details

Logic in the swap route component went out of hand, please proceed at your own risk. Hence, added the feature flag.
